src/Angor/Client/Pages/Invest.razor

```diff
@@ -520,82 +520,16 @@ else
-                                    <div class="info-label mb-2">Founder NPUB:</div>
-                                    <button class="btn btn-sm btn-border" @onclick="() => CopyToClipboard(founderNpub)">
-                                        <Icon IconName="copy" Width="16" Height="16" />
-                                    </button>
-                                </div>
-                                <div class="info-value text-break word-wrap">
-                                    @founderNpub
-                                </div>
-                            </div>
-
-                            <button type="button" class="btn btn-border-success w-100" @onclick="OpenChatApp">
-                                <Icon IconName="message" Width="20" Height="20" class="me-2" />
-                                <span>Open Chat App</span>
-                            </button>
-
-                            <button type="button" class="btn btn-border-warning w-100" @onclick="ShowNsecAndCheckPassword">
-                                <Icon IconName="key" Width="20" Height="20" class="me-2" />
-                                <span>Show NSEC</span>
-                            </button>
-
-                            @if (showNsec)
-                            {
-                                <div class="info-card p-3 mt-2 animate-fade-in">
-                                    <div class="d-flex justify-content-between align-items-center">
-                                        <div class="info-label mb-2">NSEC:</div>
-                                        <button class="btn btn-sm btn-border" @onclick="() => CopyToClipboard(nsecValue)">
-                                            <Icon IconName="copy" Width="16" Height="16" />
-                                        </button>
-                                    </div>
-                                    <div class="info-value text-break word-wrap">
-                                        @nsecValue
-                                    </div>
-                                </div>
-                            }
-                        </div>
-                    </div>
-
-                    <div class="modal-footer border-0 pt-0">
-                        <button type="button" class="btn btn-border-warning" @onclick="() => showChatModal = false">
-                            Close
-                        </button>
-                    </div>
-                </div>
-            </div>
-        </div>
-    </div>
+    <MessageComponent 
+        ContactNpub="@founderNpub"
+        ContactHexPub="@project.ProjectInfo.NostrPubKey"
+        CurrentUserNpub="@npubValue"
+        CurrentUserPrivateKeyHex="@currentUserPrivateKeyHex"
+        ChatTitle="Founder"
+        ChatAppUrl="@($"/messages/{project.ProjectInfo.ProjectIdentifier}/investor")"
+        OnClose="@(() => showChatModal = false)"
+        OnNsecRequest="@ShowNsecAndCheckPassword"
```

Comment:
what happens if the password expires in the middle of the chat popup?

Reply:
To retrieve the private key, it asks the user to enter the password again.